Sources/Scene/Assets/AutorizationStatusHeaderView.swift

```diff
@@ -0,0 +1,134 @@
+}
+
+class AutorizationStatusHeaderView : UICollectionReusableView {
+    static let id = "PHAutorizationHeaderView"
+    
+    weak var delegate: AutorizationStatusHeaderViewDelegate?
+    
+    @UsesAutoLayout
+    private var titleLabel: UILabel = {
+        let lbl = UILabel()
+        lbl.numberOfLines = 3
+        return lbl
+    }()
+    
+    @UsesAutoLayout
+    private var manageButton: UIButton = {
+        let btn = UIButton()
+        return btn
+    }()
+    
+    var buttonText: String? {
+        didSet{
+            manageButton.setTitle(buttonText, for: .normal)
+        }
+    }
+    
+    var buttonTextColor: UIColor? {
+        didSet{
+            manageButton.setTitleColor(buttonTextColor, for: .normal)
+        }
+    }
+    
+    
+    
+    var permissionDeniedText: String?
+    var permissionDeniedBackgroundColor: UIColor?
+    var permissionDeniedTextColor: UIColor?
+    
+    var limitedPermissionGrantedText: String?
+    var limitedPermissionBackgroundColor: UIColor?
+    var limitedPermissionTextColor: UIColor?
+    
+    var authorizationStatus: PHAuthorizationStatus?{
+        didSet {
+            guard let status = authorizationStatus else { return }
+            switch status {
+            case .denied:
+                titleLabel.text = permissionDeniedText
+                titleLabel.textColor = permissionDeniedTextColor
+                backgroundColor = permissionDeniedBackgroundColor
+                layoutIfNeeded()
+                break
+            case .limited:
+                titleLabel.text = limitedPermissionGrantedText
+                titleLabel.textColor = limitedPermissionTextColor
+                backgroundColor = limitedPermissionBackgroundColor
+                layoutIfNeeded()
+                break
+            default:
+                break
+            }
+        }
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        addViews()
+        constraintLayout()
+        addTargets()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func addViews(){
+        addSubview(titleLabel)
+        addSubview(manageButton)
+    }
+    
+    private func constraintLayout(){
+        NSLayoutConstraint.activate([
+            titleLabel.topAnchor.constraint(equalTo: self.topAnchor),
+            titleLabel.bottomAnchor.constraint(equalTo: self.bottomAnchor),
+            titleLabel.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 16),
+            titleLabel.trailingAnchor.constraint(lessThanOrEqualTo: self.manageButton.leadingAnchor, constant: -8),
+            
+            manageButton.topAnchor.constraint(equalTo: self.topAnchor),
+            manageButton.bottomAnchor.constraint(equalTo: self.bottomAnchor),
+            manageButton.leadingAnchor.constraint(greaterThanOrEqualTo: self.titleLabel.trailingAnchor, constant: 8),
```

Comment:
extra constraint, already created in 113 by `titleLabel.trailingAnchor.constraint(lessThanOrEqualTo: self.manageButton.leadingAnchor, constant: -8),

Reply:
Removed duplicate constraint